packages/block-library/src/navigation/index.js

```diff
@@ -52,6 +55,26 @@ export const settings = {
 	},
 	edit,
 	save,
+	__experimentalLabel: ( { ref } ) => {
+		if ( ! ref ) {
+			return __( 'Navigation' );
+		}
+
+		const navigation = select( coreStore ).getEditedEntityRecord(
+			'postType',
+			'wp_navigation',
+			ref
+		);
+
+		return navigation?.title
+			? sprintf(
+					/* translators: %1$s: block title, %2$s: navigation menu title */
+					__( '%1$s (%2$s)' ),
+					metadata.title,
+					decodeEntities( navigation.title )
+			  )
+			: __( 'Navigation' );
```

Comment:
Same here. Let's return `undefined` when Navigation has no title. You can use early return to make things more readable, IMO.